language-support/ts/daml-react/hooks.test.ts

```diff
@@ -1,31 +1,60 @@
 // Copyright (c) 2020 The DAML Authors. All rights reserved.
 // SPDX-License-Identifier: Apache-2.0
 import React, { ComponentType } from 'react';
-import { renderHook } from '@testing-library/react-hooks';
-import DamlLedger from './DamlLedger'
-import { useQuery } from './hooks';
+import { renderHook, RenderHookResult } from '@testing-library/react-hooks';
+import DamlLedger, { useQuery} from './index';
 import { Template } from '@daml/types';
+import { useParty } from './hooks';
 
-let mockResolveQuery: (contracts: string[]) => void = () => {
-  throw Error('using mockResolveQuery before init');
-}
+const mockConstructor = jest.fn();
+const mockQuery = jest.fn();
+const mockFunctions = [mockConstructor, mockQuery];
 
 jest.mock('@daml/ledger', () => class {
-  query(): Promise<string[]> {
-    return new Promise((resolve) => mockResolveQuery = resolve);
+  constructor(...args: unknown[]) {
+    mockConstructor(...args);
+  }
+  query(...args: unknown[]): Promise<string> {
+    return mockQuery(...args);
   }
 });
 
+const TOKEN = 'test_token';
+const PARTY = 'test_party';
+
+function renderDamlHook<P, R>(callback: (props: P) => R): RenderHookResult<P, R> {
+  const wrapper: ComponentType = ({children}) => React.createElement(DamlLedger, {token: TOKEN, party: PARTY}, children);
+  return renderHook(callback, {wrapper});
+}
+
 const Foo = undefined as unknown as Template<object>;
 
+beforeEach(() => {
```

Comment:
is this used anywhere?

Reply:
This registers a hook that runs before each test.

Reply:
understood now, this is called before each test.